frontend/src/routes/rankings.jsx

```diff
@@ -0,0 +1,208 @@
+      <div className="col-12 col-lg-3">
+        <Card className="mb-2">
+          <Card.Body className="p-2">
+            <Card.Title className="dataelem text-truncate">Rankings</Card.Title>
+            <Card.Text className="small">{showDate()}</Card.Text>
+            <Card.Text>
+              <Form.Control type="date" value={readDate()} onChange={handleChange} size="sm" />
+            </Card.Text>
+          </Card.Body>
+        </Card>
+        <Button as={Link} to={readLink("d")} variant="secondary" className="d-grid mb-2" size="sm">
+          For today
+        </Button>
+        <Button as={Link} to={readLink("w")} variant="secondary" className="d-grid mb-2" size="sm">
+          This week
+        </Button>
+        <Button as={Link} to={readLink("m")} variant="secondary" className="d-grid mb-2" size="sm">
+          This month
+        </Button>
+        <Button as={Link} to={readLink("y")} variant="secondary" className="d-grid mb-2" size="sm">
+          This year
+        </Button>
+        <Button as={Link} to={readLink()} variant="secondary" className="d-grid mb-2" size="sm">
+          All time
+        </Button>
+        {scanDate() ? <hr className="mt-0 mb-2" /> : null}
+        {params.d && scanDate() && !isweekly ? (
+          <Button
+            as={Link}
+            to={`/rankings/y/${params.y}/m/${params.m}/d/${params.d}/week`}
+            variant="secondary"
+            className="d-grid mb-2"
+            size="sm"
+          >
+            That week
+          </Button>
+        ) : null}
+        {params.d && scanDate() && isweekly ? (
+          <Button
+            as={Link}
+            to={`/rankings/y/${params.y}/m/${params.m}/d/${params.d}`}
+            variant="secondary"
+            className="d-grid mb-2"
+            size="sm"
+          >
+            That date
+          </Button>
+        ) : null}
+        {params.m && scanDate() ? (
+          <Button
+            as={Link}
+            to={`/rankings/y/${params.y}/m/${params.m}`}
+            variant="secondary"
+            className="d-grid mb-2"
+            size="sm"
+          >
+            For {new Date(params.y, params.m - 1).toLocaleDateString("en-US", { month: "long" })}
+          </Button>
+        ) : null}
+        {params.y && scanDate() ? (
+          <Button as={Link} to={`/rankings/y/${params.y}`} variant="secondary" className="d-grid" size="sm">
+            For {params.y}
+          </Button>
+        ) : null}
+      </div>
+      <div className="col-12 col-lg-9">
+        <Card className="mb-2">
+          <Card.Body className="ps-0 pe-0 pt-2 pb-0">
+            <Card.Title className="mb-0 ps-2 dataelem" style={{ textTransform: "capitalize" }}>
+              {showHead()}
+            </Card.Title>
+            <Card.Text className="mb-0 ps-2 small">Showing top performers</Card.Text>
+            <hr className="mt-2 mb-0" />
+            <ListGroup variant="flush">
+              {dict && dict.length > 0 ? (
```

Comment:
What happens if `length` is zero? I think it won't produce any output already (but I may be wrong).

Reply:
This, if I understood your question correction https://github.com/fedora-infra/tahrir/pull/703/files#diff-e25eb7f4400fd68a3a86343afb86341009b329b8f19a3a51e3facb83f0b42087R200. 

<img width="1280" height="500" alt="image" src="https://github.com/user-attachments/assets/336cd068-2089-438b-9ac8-0caf89c23038" />